### ISSUE TITLE
Downloads don't reserve space for preventing not download

### ISSUE THREAD

**Issue by animekai** [mmozeiko/pkgi#9]
i have this problem when i download a game, for example i had 3 gb  of free space, a game size is 2.50 gb, when is 50% on download and i lost connection, when i want to resume the download, the program said there is no free space Available, but thats not true, the space is occupied by the same download in progress, but it take reference from the free space Available.

i think pkgi must reserve space before start downloading, because of this issue, resuming downloads with mininal enought space, is causing problems.

**Comment by mmozeiko** [comment]
You are right. It should have extra check when download is resumed. I'll fix it.

I don't think there is a reasonable way to reserve space before starting download on Vita. It would require writing each file full of zeroes or something. Which could take a lot of time - sdcard access is not fast.

**Comment by animekai** [comment]
yeah because is a problem when trying to daonload a game with just the minimal space left, if for some reason the download stops, the free espace available prevents to resume the download, and it forces you to erase manually the incomplete data just to start from again from the beginning, is anooying XD.

**Comment by CelesteBlue-dev** [comment]
you should do like SONY : they set the file size to the final size, even tough written size is currently inferior.
I don't know how they do that... But I think yes they certainly write the temp file full of zeroes then erase it.

**Comment by SilicaAndPina** [comment]
Why not just use the offical download system, then it would work as a background download and basicly do all the queing work for you..

**Comment by mmozeiko** [comment]
Please explain how to implement it (what are the API's or maybe sample code)  and I'll do that right away.

**Comment by SilicaAndPina** [comment]
https://github.com/TheOfficialFloW/DownloadEnabler/blob/master/main.c ?

**Comment by mmozeiko** [comment]
As already discussed in #4 it only downloads, not unpacks - meaning 2x space usage on flash. Also it requires manual click to start download and it requires switching between applications when download starts or fiinishes, which it is terrible experience for user.

**Comment by SilicaAndPina** [comment]
hmm IDU Settings "Content Downloader" is probs where you need to look then (it downloads and unpacks PKG basicly offically)

**Comment by mmozeiko** [comment]
I'm not familiar with that. What is IDU settings "Content Downloader"? Can you provide some links?

**Comment by SilicaAndPina** [comment]
IDU Settings is a hidden settings menu on IDU / Kiosk mode vitas, content downloader is one of the options in the settings menu

**Comment by SilicaAndPina** [comment]
IDU Settings is a hidden settings menu in IDU / Kiosk mode vita's and content downloader is one of the options in it.

**Comment by animekai** [comment]
what about implementing not just vita games/dlc, what about psm games, psp/ps1/classics games too, all the available library of digital games from sony servers for psvita in pkgi it would be awesome.

**Comment by mmozeiko** [comment]
> but idk if it works..

Well that's the answer to your original question "Why not just use the offical download system" - idk how it works :)

> what about psm games, psp/ps1/classics games too,

Pull requests accepted.

And please create separate issue(s). This is not some random forum. This is issue about checking free space for interrupted downloads. I will delete further off-topic comments.

**Comment by SilicaAndPina** [comment]
hmm i might look into the official system sometime .-.
